dataland-backend/src/test/kotlin/org/dataland/datalandbackend/annotations/DataTypesExtractorTest.kt

```diff
@@ -7,10 +7,10 @@ class DataTypesExtractorTest {
 
     @Test
     fun `check if extracting data types works as expected`() {
-        val expectedTypes = listOf("EuTaxonomyData")
+        val expectedTypes = listOf("EuTaxonomyDataForFinancials", "EuTaxonomyDataForNonFinancials")
         val dataTypes = DataTypesExtractor().getAllDataTypes()
         Assertions.assertTrue(
-            dataTypes == expectedTypes,
+            dataTypes.containsAll(expectedTypes),
```

Comment:
Test it the other way around?

Reply:
Test stellt sicher, dass alle eventuell neu hinzugekommenen FrameworkTypes korrekt annotiert sind.
Das ist wichtig, damit der DataManager im Backend bei der Methode `verifyDataTypeExists(dataType: String)` ein korrektes Ergebnis liefert.

Reply:
Die Änderung in containsAll erscheint mir nicht sinnvoll. Mein Verständnis von diesem Test ist das geprüft werden soll das genau die erwarteten types da sind. Mit containsAll wird geprüft ob die vorhandenen dataTypes als Untermenge die expectedTypes enthalten. Falls das so beabsichtigt ist würde ich gerne die Motivation dahinter verstehen.

Reply:
Das war nicht so beabsichtigt.
Das "==" vorher war leider auch nicht ganz treffend, weil es die Reihenfolge der Listen auch verglichen hat:
D.h. expected type [TypA, TypB] wäre ungleich datatypes [TypB, TypA].

Wenn ich jetzt so darüber nachdenke, wäre es eigentlich das beste, wenn man hier alles als sets behandelt, und dann im assert abprüft, ob die Sets EXAKT die selben Elemente enthalten.
Das würde alles erschlagen.

Dazu könnte man auch gerne den dataTypeExtractor im Backend, der die Typen über die Annotations alle sammelt, so umschreiben, dass er sein Ergebnis als Set of Strings speichert, und nicht mehr als List of Strings.

Reply:
Alternativ könnte man auch die beiden Listen sortieren

Reply:
Frech gefragt: Wenn wir diesen Test immer updaten, was bringt uns dann unsere Reflection-Klasse. Können wir dann nicht direkt einfach ne Konstante Liste ins Backend schreiben?. Deswegen finde ich das containsAll mit diesen beiden Beispielen (und vielleicht ein negativbeispiel, sodass nicht jede Klasse in der Liste landet) voll ausreichend.

Reply:
Die Idee von diesem Test war sicherzustellen das die von Reflektion gelieferten Klassen den erwarteten entsprechen. Dafür muss ich ja das Erwartungsbild hinschreiben. Die Frage ob man sich das ganze Reflektionzeug hätte sparen können und stattdessen einfach eine manuell gepflegte Liste hätte verwenden können, ist ein anderes Thema.